backend/store/policy.go

```diff
@@ -5,33 +5,152 @@ import (
 	"database/sql"
 	"fmt"
 	"log/slog"
+	"slices"
 	"strings"
 	"time"
 
 	"github.com/pkg/errors"
+	"google.golang.org/protobuf/encoding/protojson"
 
 	"github.com/bytebase/bytebase/backend/common"
 	"github.com/bytebase/bytebase/backend/common/log"
 	api "github.com/bytebase/bytebase/backend/legacyapi"
 	storepb "github.com/bytebase/bytebase/proto/generated-go/store"
 )
 
-func (s *Store) GetProjectIamPolicy(ctx context.Context, projectUID int) (*storepb.ProjectIamPolicy, error) {
+func (s *Store) GetWorkspaceIamPolicy(ctx context.Context) (*storepb.IamPolicy, error) {
```

Comment:
Wanna use the following interface once we can get rid of roles from user/principal proto.

```
	if _, err := s.store.CreatePolicyV2(ctx, &store.PolicyMessage{
		ResourceUID:       project.UID,
		ResourceType:      api.PolicyResourceTypeProject,
		Payload:           string(policyPayload),
		Type:              api.PolicyTypeProjectIAM,
		InheritFromParent: false,
		// Enforce cannot be false while creating a policy.
		Enforce: true,
	}, creatorUID); err != nil {
		return nil, status.Errorf(codes.Internal, err.Error())
	}
```